packages/components/badge/src/Badge.vue

```diff
@@ -0,0 +1,78 @@
+<template>
+  <span class="ix-badge-wrapper">
+    <slot />
+    <sup class="ix-badge" :class="classes" :style="styles">
+      <slot v-if="$slots.custom" name="custom" />
+      <template v-else-if="countValue">{{ countValue }}</template>
+    </sup>
+  </span>
+</template>
+<script lang="ts">
+import { computed, defineComponent, Slots } from 'vue'
+import { BadgeProps } from './types'
+import { isNumeric, isDefined } from './utils'
+
+export default defineComponent({
+  name: 'IxBadge',
+  props: {
+    dot: { type: Boolean, default: false },
+    count: { type: [Number, String], default: 0 },
+    overflowCount: { type: [Number, String], default: 99 },
+    showZero: { type: Boolean, default: false },
+    color: { type: String, default: undefined },
+  },
+  setup(props: BadgeProps, { slots }) {
+    const countValue = useCountValue(props, slots)
+
+    const classes = useClasses(props, slots)
+
+    const styles = useStyles(props, slots)
+
+    return {
+      countValue,
+      classes,
+      styles,
+    }
+  },
+})
+
+const useCountValue = (props: BadgeProps, slots: Slots) => {
+  return computed(() => {
+    if (!slots.custom && !props.dot) {
+      if (!props.showZero && +props.count === 0) return false
+      if (isNumeric(props.count) && isNumeric(props.overflowCount)) {
+        return +props.count > +props.overflowCount ? `${props.overflowCount}+` : `${props.count}`
+      }
+      return props.count
+    }
+    return false
+  })
+}
+
+const useStyles = (props: BadgeProps, slots: Slots) => {
+  return computed(() => {
+    const { custom: customSlot } = slots
+    const color = isDefined(props.color) && props.color
```

Comment:
`isDefined` 这个工具函数是否有必要？我感觉下面这种写法更直观。
```suggestion
    const color = props.color ?? false
```